### ISSUE TITLE
Error when switching institutions while a box is being displayed

### ISSUE THREAD

**Issue by diegoliberman** [instedd/cdx#1713]

**Comment by straight-shoota** [comment]
I'm pretty sure you'll see the same behaviour with other resource type as well.

**Comment by ysbaddaden** [comment]
That could be fixed with a custom 404 Not Found page, where we'd render the page layout, so we're not struck on an error page.

**Comment by diegoliberman** [comment]
> I'm pretty sure you'll see the same behaviour with other resource type as well.

No, it doesn't.  I tested with Samples and Batches.  There's no error there, but probably the UX isn't the best: you remain in a inconsistent situation where you have 1 institution selected, but you're seeing a record from another institution.

**Comment by straight-shoota** [comment]
Boxes works like this as well.

> you remain in a inconsistent situation where you have 1 institution selected, but you're seeing a record from another institution.

IMO that's a plain bug. And it's potentially worse because access checks might not be working properly. As long as the user has legit access to the sample via a different institution, it's just a UX issue. But when the sample is not properly checked against the current navigation context, there might be more relevant security issues.

**Comment by diegoliberman** [comment]
> IMO that's a plain bug. And it's potentially worse because access checks might not be working properly. As long as the user has legit access to the sample via a different institution, it's just a UX issue. But when the sample is not properly checked against the current navigation context, there might be more relevant security issues. (that's pure speculation, I didn't look at any code)

It's just an UX issue.  I would consider a general solution such as that we always navigate to the Home when the user changes the Institution.

**Comment by ysbaddaden** [comment]
It could be nice to redirect to the list of resources instead of the home (e.g. `/samples` or `/tests`).
